waspc/data/Generator/templates/react-app/src/actions/index.js

```diff
@@ -0,0 +1,74 @@
+import {
+  useMutation,
+  useQueryClient,
+} from 'react-query'
+
+export { configureQueryClient } from '../queryClient'
+
+export function useAction(actionFn, actionOptions) {
+  const queryClient = useQueryClient();
+
+  // TODO(filip): Do we allow all react query options or should we consider a list of allowed options?
+  let options = {}
+
+  if (actionOptions?.optimisticUpdates) {
+    const optimisticUpdatesConfig = makeOptimisticUpdatesConfig(actionOptions.optimisticUpdates)
+    options = makeOptimisticUpdateOptions(queryClient, optimisticUpdatesConfig)
+  }
```

Comment:
You could make this a bit more immutable/functional like this:

```js
const options = actionOptions?.optimisticUpdates
  ? makeOptimisticUpdateOptions(
      queryClient,  
      makeOptimisticUpdatesConfig(actionOptions.optimisticUpdates)
    )
  : {} 
```

Reply:
I would also consider renaming them to `mutationOptions`.

Reply:
In the sole context of this PR: yes, it makes the code nicer.

However, both PRs/approaches that the flickering problem assign an additional thing inside the if block (a modified `mutationFn`). In those cases, we'd need either:
- Two identical ternary operators and two separate calls to `makeOptimisticUpdatesConfig` for two assignments
- Three identical ternary operators for three assignments and no way to communicate that they're mutually dependent (which would be a problem if we had types)

So, in both PRs that build on this one, the cleaner approach is using an `if` statement. I decided to leave the `if` statement in this (base) PR as well. It avoids cluttering the diff with irrelevant/cosmetic changes and keeps the spotlight on the changes required to solve the flickering UI problem.